addon/components/polaris-resource-list/bulk-actions.js

```diff
@@ -0,0 +1,535 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { throttle } from '@ember/runloop';
+import { warn } from '@ember/debug';
+import ContextBoundEventListenersMixin from 'ember-lifeline/mixins/dom';
+import layout from '../../templates/components/polaris-resource-list/bulk-actions';
+
+const MAX_PROMOTED_ACTIONS = 2;
+
+export default Component.extend(ContextBoundEventListenersMixin, {
+  layout,
+
+  /**
+   * Visually hidden text for screen readers
+   *
+   * @type {String}
+   * @default null
+   * @property accessibilityLabel
+   */
+  accessibilityLabel: null,
+
+  /**
+   * Label for the bulk actions
+   *
+   * @type {String}
+   * @default ''
+   * @property label
+   */
+  label: '',
+
+  /**
+   * State of the bulk actions checkbox
+   *
+   * @type {Boolean|String}
+   * @default null
+   * @property selected
+   */
+  selected: null,
+
+  /**
+   * List is in a selectable state
+   *
+   * @type {Boolean}
+   * @default false
+   * @property selectMode
+   */
+  selectMode: false,
+
+  /**
+   * Actions that will be given more prominence
+   *
+   * @type {Object[]}
+   * @default null
+   * @property promotedActions
+   */
+  promotedActions: null,
+
+  /**
+   * List of actions
+   *
+   *
+   * @type {Object[]}
+   * @default null
+   * @property actionsCollection
+   */
+  actionsCollection: null,
+
+  /**
+   * Text to select all across pages
+   *
+   * @type {String}
+   * @default null
+   * @property paginatedSelectAllText
+   */
+  paginatedSelectAllText: null,
+
+  /**
+   * Action for selecting all across pages
+   *
+   * @type {Function}
+   * @default null
+   * @property paginatedSelectAllAction
+   */
+  paginatedSelectAllAction: null,
+
+  /**
+   * Disables bulk actions
+   *
+   * @type {Boolean}
+   * @default null
+   * @property disabled
+   */
+  disabled: null,
+
+  /**
+   * @private
```

Comment:
I haven't been tagging internal/private props like this as the linter considers them private by default and it just adds clutter - thoughts?

Reply:
Yeah I'm good with removing 👍